integrations/access/opsgenie/app.go

```diff
@@ -357,6 +408,42 @@ func (a *App) tryNotifyService(ctx context.Context, req types.AccessRequest) (bo
 	return isNew, nil
 }
 
+func (a *App) getMessageRecipients(ctx context.Context, req types.AccessRequest) ([]common.Recipient, []common.Recipient, error) {
+	recipientSetSchedules := common.NewRecipientSet()
+	recipientSchedules := a.accessMonitoringRules.RecipientsFromAccessMonitoringRules(ctx, req)
+	recipientSchedules.ForEach(func(r common.Recipient) {
+		recipientSetSchedules.Add(r)
+	})
+	// Access Monitoring Rules recipients does not have a way to handle separate recipient types currently.
+	// Recipients from Access Monitoring Rules will be schedules only currently.
+	if recipientSetSchedules.Len() != 0 {
+		return recipientSetSchedules.ToSlice(), nil, nil
+	}
+	rawSchedules, rawTeams, err := a.getNotifyServiceNames(ctx, req)
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+	for _, rawSchedule := range rawSchedules {
+		rec := &common.Recipient{
+			Name: rawSchedule,
+			ID:   rawSchedule,
+			Kind: common.RecipientKindSchedule,
+		}
+		recipientSetSchedules.Add(*rec)
```

Comment:
Why is the pointer necessary?

```suggestion
		recipientSetSchedules.Add(common.Recipient{
			Name: rawSchedule,
			ID:   rawSchedule,
			Kind: common.RecipientKindSchedule,
		})
```